### ISSUE TITLE
compiling problem

### ISSUE THREAD

**Issue by Charliechen1** [baidu/AnyQ#5]
Hi,
I am comping the project but I have got the following error:

> [  4%] Built target extern_leveldb
[  9%] Built target extern_jsoncpp
[ 13%] Built target extern_gtest
[ 18%] Built target extern_xgboost
[ 22%] Built target extern_eigen
[ 23%] Performing build step for 'extern_paddle'
[  1%] Built target extern_lib_any
[  2%] Built target extern_eigen3
[  2%] Built target extern_gflags
[  4%] Built target extern_threadpool
[  4%] Built target extern_snappy
[  4%] Built target extern_mklml
[  4%] Built target extern_zlib
[  5%] Built target extern_boost
[  6%] Built target profiler_py_proto_init
[  6%] Built target extern_warpctc
[  6%] Built target framework_py_proto_init
[  7%] Built target extern_glog
[  7%] Built target extern_snappystream
[  8%] Built target extern_protobuf
[  8%] Built target stringpiece
[  9%] Built target header
[  9%] Built target enforce
[  9%] Built target cblas
[ 10%] Built target memory_block
Copy generated python proto into directory paddle/fluid/proto.
[ 10%] Built target activation_functions
[ 10%] Generating ../paddle/python/paddle/proto/ParameterService_pb2.py
[ 10%] Generating ../paddle/python/paddle/proto/ParameterServerConfig_pb2.py
[ 10%] Generating ../paddle/python/paddle/proto/TrainerConfig_pb2.py
Copy generated python proto into directory paddle/fluid/proto/profiler.
[ 10%] Generating ../paddle/python/paddle/proto/OptimizerConfig_pb2.py
[ 10%] Generating ../paddle/python/paddle/proto/ParameterConfig_pb2.py
[ 10%] Generating ../paddle/python/paddle/proto/ModelConfig_pb2.py
[ 11%] Generating ../paddle/python/paddle/proto/DataFormat_pb2.py
[ 11%] Generating ../paddle/python/paddle/proto/DataConfig_pb2.py
[ 11%] Built target framework_proto
[ 12%] Built target paddle_proto
[ 12%] Built target chunk
[ 12%] Built target framework_py_proto
[ 12%] Built target ddim
[ 12%] Built target cpu_info
[ 12%] Built target profiler_py_proto
[ 13%] Built target dynamic_loader
[ 13%] Built target gen_proto_py
[ 15%] Built target threadpool
[ 15%] Built target prune
[ 16%] Built target writer
[ 16%] Built target place
[ 17%] Built target profiler_proto
[ 17%] Built target system_allocator
[ 17%] Built target scanner
[ 17%] Built target dynload_warpctc
[ 17%] Built target attribute
[ 17%] Built target scope
[ 18%] Built target buddy_allocator
[ 19%] Built target recordio
[ 19%] Built target var_handle
[ 19%] Built target memcpy
[ 19%] Built target device_tracer
[ 19%] Built target op_proto_maker
[ 19%] Built target op_info
[ 19%] Built target malloc
[ 19%] Built target memory
[ 19%] Built target device_context
[ 19%] Built target vol2col
[ 19%] Built target data_type
[ 20%] Built target sequence2batch
[ 20%] Built target maxouting
[ 20%] Built target sequence_scale
[ 20%] Built target blas
[ 20%] Built target concurrency
[ 20%] Built target pooling
[ 20%] Built target im2col
[ 20%] Built target lstm_compute
[ 20%] Built target sequence_padding
[ 20%] Built target profiler
[ 20%] Built target concat
[ 20%] Built target cos_sim_functor
[ 21%] Built target shape_inference
[ 21%] Built target cross_entropy
[ 22%] Built target unpooling
[ 23%] Built target math_function
[ 23%] Built target tensor
[ 23%] Built target softmax
[ 23%] Built target gru_compute
[ 23%] Built target data_type_transform
[ 23%] Built target data_layout_transform
[ 23%] Built target selected_rows
[ 24%] Built target lod_tensor
[ 24%] Built target context_project
[ 24%] Built target sequence_pooling
[ 25%] Built target data_device_transform
[ 25%] Built target op_handle_base
[ 26%] Built target feed_fetch_method
[ 26%] Built target variable_visitor
[ 26%] Built target reader
[ 26%] Built target lod_rank_table
[ 26%] Built target selected_rows_functor
[ 26%] Built target data_transform
[ 26%] Built target ssa_graph
[ 26%] Built target fetch_op_handle
[ 26%] Built target fuse_vars_op_handle
[ 26%] Built target broadcast_op_handle
[ 26%] Built target reduce_op_handle
[ 26%] Built target all_reduce_op_handle
[ 27%] Built target gather_op_handle
[ 28%] Built target operator
[ 28%] Built target scale_loss_grad_op_handle
[ 29%] Built target ssa_graph_builder
[ 29%] Built target ssa_graph_executor
[ 29%] Built target init
[ 29%] Built target proto_desc
[ 29%] Built target ssa_graph_printer
[ 29%] Built target threaded_ssa_graph_executor
[ 29%] Built target ssa_graph_checker
[ 29%] Built target scope_buffered_ssa_graph_executor
[ 29%] Built target op_registry
[ 29%] Built target scale_op
[ 29%] Built target top_k_op
[ 29%] Built target sigmoid_cross_entropy_with_logits_op
[ 29%] Built target shape_op
[ 29%] Built target sequence_concat_op
[ 30%] Built target executor
[ 30%] Built target smooth_l1_loss_op
[ 31%] Built target computation_op_handle
[ 32%] Built target split_lod_tensor_op
[ 32%] Built target roi_pool_op
[ 33%] Built target rpc_op_handle
[ 33%] Built target uniform_random_op
[ 33%] Built target split_ids_op
[ 33%] Built target split_op
[ 33%] Built target tensor_array_read_write_op
[ 33%] Built target rnn_memory_helper_op
[ 33%] Built target rmsprop_op
[ 33%] Built target reverse_op
[ 33%] Built target select_op
[ 33%] Built target precision_recall_op
[ 34%] Built target reorder_lod_tensor_by_rank_op
[ 34%] Built target reduce_sum_op
[ 34%] Built target norm_op
[ 34%] Built target prelu_op
[ 34%] Built target rank_loss_op
[ 35%] Built target proximal_gd_op
[ 35%] Built target positive_negative_pair_op
[ 35%] Built target split_byref_op
[ 35%] Built target pad_op
[ 35%] Built target sign_op
[ 35%] Built target modified_huber_loss_op
[ 35%] Built target nce_op
[ 35%] Built target minus_op
[ 36%] Built target mean_op
[ 37%] Built target scatter_op
[ 37%] Built target merge_ids_op
[ 37%] Built target mean_iou_op
[ 37%] Built target merge_lod_tensor_op
[ 37%] Built target lod_array_length_op
[ 37%] Built target margin_rank_loss_op
[ 37%] Built target lod_reset_op
[ 37%] Built target log_loss_op
[ 37%] Built target split_selected_rows_op
[ 37%] Built target logical_op
[ 37%] Built target lstm_unit_op
[ 37%] Built target row_conv_op
[ 37%] Built target linear_chain_crf_op
[ 38%] Built target lrn_op
[ 38%] Built target layer_norm_op
[ 38%] Built target spp_op
[ 39%] Built target adamax_op
[ 39%] Built target average_accumulates_op
[ 39%] Built target activation_op
[ 40%] Built target load_combine_op
[ 40%] Built target edit_distance_op
[ 40%] Built target channel_recv_op
[ 41%] Built target channel_send_op
[ 41%] Built target sequence_erase_op
[ 41%] Built target arg_min_op
[ 41%] Built target clip_op
[ 41%] Built target adadelta_op
[ 41%] Built target save_op
[ 41%] Built target save_combine_op
[ 41%] Built target dropout_op
[ 42%] Built target decayed_adagrad_op
[ 42%] Built target detection_map_op
[ 42%] Built target huber_loss_op
[ 42%] Built target fill_constant_batch_size_like_op
[ 42%] Built target while_op
[ 42%] Built target conv_op
[ 42%] Built target warpctc_op
[ 42%] Built target sequence_slice_op
[ 43%] Built target parallel_do_op
[ 43%] Built target elementwise_max_op
[ 43%] Built target softmax_op
[ 43%] Built target batch_norm_op
[ 43%] Built target sequence_conv_op
[ 43%] Built target lookup_sparse_table_op
[ 43%] Built target adam_op
[ 43%] Built target cumsum_op
[ 43%] Built target accuracy_op
[ 43%] Built target cos_sim_op
[ 43%] Built target transpose_op
[ 43%] Built target feed_op
[ 44%] Built target sum_op
[ 44%] Built target reduce_mean_op
[ 44%] Built target max_sequence_len_op
[ 45%] Built target reduce_max_op
[ 45%] Built target sequence_softmax_op
[ 45%] Built target gru_op
[ 45%] Built target l1_norm_op
[ 45%] Built target maxout_op
[ 45%] Built target conv_transpose_op
[ 45%] Built target ctc_align_op
[ 45%] Built target cross_entropy_op
[ 45%] Built target cast_op
[ 45%] Built target auc_op
[ 45%] Built target gaussian_random_batch_size_like_op
[ 45%] Built target mul_op
[ 45%] Built target softmax_with_cross_entropy_op
[ 46%] Built target sequence_pool_op
[ 46%] Built target lookup_table_op
[ 46%] Built target expand_op
[ 46%] Built target recurrent_op
[ 48%] Built target assign_value_op
[ 48%] Built target momentum_op
[ 48%] Built target label_smooth_op
[ 48%] Built target is_empty_op
[ 48%] Built target arg_max_op
[ 48%] Built target print_op
[ 48%] Built target lod_rank_table_op
[ 49%] Built target one_hot_op
[ 49%] Built target concat_op
[ 49%] Built target lstm_op
[ 49%] Built target assign_op
[ 49%] Built target load_op
[ 49%] Built target channel_create_op
[ 49%] Built target sequence_expand_op
[ 49%] Built target pool_op
[ 49%] Built target squared_l2_distance_op
[ 49%] Built target sequence_reshape_op
[ 49%] Built target elementwise_pow_op
[ 49%] Built target sgd_op
[ 49%] Built target fake_dequantize_op
[ 49%] Built target random_crop_op
[ 49%] Built target beam_search_decode_op
[ 49%] Built target elementwise_sub_op
[ 49%] Built target pool_with_index_op
[ 50%] Built target beam_search_op
[ 50%] Built target bilinear_interp_op
[ 50%] Built target slice_op
[ 50%] Built target unpool_op
[ 50%] Built target get_places_op
[ 50%] Built target elementwise_add_op
[ 50%] Built target adagrad_op
[ 50%] Built target shrink_rnn_memory_op
[ 50%] Built target channel_close_op
[ 50%] Built target clip_by_norm_op
[ 50%] Built target fill_zeros_like_op
[ 50%] Built target chunk_eval_op
[ 50%] Built target multiplex_op
[ 50%] Built target compare_op
[ 50%] Built target conditional_block_op
[ 50%] Built target bilinear_tensor_product_op
[ 50%] Built target read_op
[ 50%] Built target lstmp_op
[ 50%] Built target crf_decoding_op
[ 50%] Built target reduce_prod_op
[ 50%] Built target crop_op
[ 50%] Built target gru_unit_op
[ 51%] Built target conv_shift_op
[ 51%] Built target proximal_adagrad_op
[ 51%] Built target increment_op
[ 51%] Built target uniform_random_batch_size_like_op
[ 51%] Built target delete_var_op
[ 51%] Built target fill_constant_op
[ 52%] Built target elementwise_mul_op
[ 52%] Built target gather_op
[ 52%] Built target elementwise_min_op
[ 52%] Built target elementwise_div_op
[ 52%] Built target squared_l2_norm_op
[ 52%] Built target ftrl_op
[ 53%] Built target fill_op
[ 53%] Built target reshape_op
[ 53%] Built target reduce_min_op
[ 53%] Built target gaussian_random_op
[ 53%] Built target fetch_op
[ 53%] Built target hinge_loss_op
[ 53%] Built target im2sequence_op
[ 53%] Built target matmul_op
[ 54%] Built target iou_similarity_op
[ 54%] Built target go_op
[ 54%] Built target target_assign_op
[ 54%] Built target reader_op_registry
[ 54%] Built target mine_hard_examples_op
[ 54%] Built target polygon_box_transform_op
[ 54%] Built target multiclass_nms_op
[ 54%] Built target bipartite_match_op
[ 54%] Built target prior_box_op
[ 54%] Built target box_coder_op
[ 54%] Built target multi_devices_graph_builder
[ 54%] Built target lod_tensor_to_array_op
[ 54%] Built target array_to_lod_tensor_op
[ 54%] Built target create_recordio_file_reader_op
[ 54%] Built target create_shuffle_reader_op
[ 54%] Built target open_files_op
[ 55%] Built target create_threaded_reader_op
[ 55%] Built target create_random_data_generator_op
[ 55%] Built target create_double_buffer_reader_op
[ 55%] Built target create_multi_pass_reader_op
[ 55%] Built target create_batch_reader_op
[ 56%] Built target create_custom_reader_op
[ 56%] Built target ssa_graph_builder_factory
[ 56%] Built target paddle_fluid_api
[ 56%] Built target parallel_executor
make[5]: *** No rule to make target `third_party/install/zlib/lib/libz.a', needed by `paddle/fluid/inference/libpaddle_fluid.so'.  Stop.
make[4]: *** [paddle/fluid/inference/CMakeFiles/paddle_fluid_shared.dir/all] Error 2
make[4]: *** Waiting for unfinished jobs....
[100%] Built target paddle_fluid
make[3]: *** [all] Error 2
make[2]: *** [third_party/paddle/src/extern_paddle-stamp/extern_paddle-build] Error 2
make[1]: *** [CMakeFiles/extern_paddle.dir/all] Error 2
make: *** [all] Error 2

What shall I do to solve it? Thank you.

**Comment by zhanghan1992** [comment]
What are the versions of your gcc and cmake ? The building log you provided is only printed in 'stdout'.  We also need to see the log printed in 'stderr' so that we can  track and locate this error.  In order for us to see the complete log， you can build with the instrunction 'make 1>stderr.log 2>&1 ', and send the file 'stderr.log' to zhanghan17@baidu.com. Thank you!

**Comment by Charliechen1** [comment]
my gcc is 4.8.5 and my cmake is 3.6.2

**Comment by liqinc** [comment]
hi,i also have this problem ,how can i solve it? Thank you!

**Comment by luotao1** [comment]
What's your cmake command and compile command?
> No rule to make target third_party/install/zlib/lib/libz.a

Maybe the network is not good to download the zlib library.

**Comment by Charliechen1** [comment]
Ummmm, seems I did what exactly the readme file asked me to do.

**Comment by zhanghan1992** [comment]
Please make sure that the source code of zlib is on the path 'build/third_party/install/paddle/third_party/zlib/src/extern_zlib'. And if not, the error might be caused by network problem.

**Comment by zhanghan1992** [comment]
Your gcc is missing yacc. Install yaccc before build this project.

**Comment by erhaixilu** [comment]
Thank you very much.it's ok.Execute the following command
-----
yum install byacc
yum install flex
